docs/advanced-ai/intro-tutorial.md

```diff
@@ -1,217 +1,127 @@
6d-f201-4378-8f66-ce9a6cfd5f3b%22%2C%0A%20%20%20%20%20%20%22name%22%3A%20%22OpenAI%20Chat%20Model%22%2C%0A%20%20%20%20%20%20%22credentials%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%22openAiApi%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22id%22%3A%20%22jiPPcYV9I70iKapN%22%2C%0A%20%20%20%20%20%20%20%20%20%20%22name%22%3A%20%22OpenAi%20account%2037%22%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%22parameters%22%3A%20%7B%7D%2C%0A%20%20%20%20%20%20%22type%22%3A%20%22%40n8n%2Fn8n-nodes-langchain.memoryBufferWindow%22%2C%0A%20%20%20%20%20%20%22typeVersion%22%3A%201.3%2C%0A%20%20%20%20%20%20%22position%22%3A%20%5B%0A%20%20%20%20%20%20%20%2020%2C%0A%20%20%20%20%20%20%20%20180%0A%20%20%20%20%20%20%5D%2C%0A%20%20%20%20%20%20%22id%22%3A%20%22afbab05c-1e87-4f7a-9d66-c86f9db1ec64%22%2C%0A%20%20%20%20%20%20%22name%22%3A%20%22Window%20Buffer%20Memory%22%0A%20%20%20%20%7D%0A%20%20%5D%2C%0A%20%20%22pinData%22%3A%20%7B%7D%2C%0A%20%20%22connections%22%3A%20%7B%0A%20%20%20%20%22When%20chat%20message%20received%22%3A%20%7B%0A%20%20%20%20%20%20%22main%22%3A%20%5B%0A%20%20%20%20%20%20%20%20%5B%0A%20%20%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22node%22%3A%20%22AI%20Agent%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22type%22%3A%20%22main%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22index%22%3A%200%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%20%20%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%22OpenAI%20Chat%20Model%22%3A%20%7B%0A%20%20%20%20%20%20%22ai_languageModel%22%3A%20%5B%0A%20%20%20%20%20%20%20%20%5B%0A%20%20%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22node%22%3A%20%22AI%20Agent%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22type%22%3A%20%22ai_languageModel%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22index%22%3A%200%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%20%20%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%22Window%20Buffer%20Memory%22%3A%20%7B%0A%20%20%20%20%20%20%22ai_memory%22%3A%20%5B%0A%20%20%20%20%20%20%20%20%5B%0A%20%20%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22node%22%3A%20%22AI%20Agent%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22type%22%3A%20%22ai_memory%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%22index%22%3A%200%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%5D%0A%20%20%20%20%20%20%5D%0A%20%20%20%20%7D%0A%20%20%7D%2C%0A%20%20%22active%22%3A%20false%2C%0A%20%20%22settings%22%3A%20%7B%0A%20%20%20%20%22executionOrder%22%3A%20%22v1%22%0A%20%20%7D%2C%0A%20%20%22versionId%22%3A%20%22b1641385-c6b0-48a8-8e26-20d1f6bd7fda%22%2C%0A%20%20%22meta%22%3A%20%7B%0A%20%20%20%20%22templateCredsSetupCompleted%22%3A%20true%2C%0A%20%20%20%20%22instanceId%22%3A%20%22cb484ba7b742928a2048bf8829668bed5b5ad9787579adea888f05980292a4a7%22%0A%20%20%7D%2C%0A%20%20%22id%22%3A%20%22l05TkWXXYbOiuL4o%22%2C%0A%20%20%22tags%22%3A%20%5B%5D%0A%7D" frame="true">
```

Comment:
This workflow uses the frame so it can be easily copied (if someone made a mistake or couldn't get those theirs to work. Could this work as a template instead?

Reply:
By "as a template" do you mean published [here](https://n8n.io/workflows/) or something else?

Reply:
I'm not sure how the distinction works, but when using a template, rather than importing, you get a splash screen indicating what things you need to add (credentials), which I think would be better for this usage

Reply:
Got it, yeah. For templates published in that library, we have a [separate function that renders those](https://github.com/n8n-io/n8n-docs/blob/4ef2977ba49c7ba02c5b4159f44eb23be453fd3d/main.py#L35-L85), which is a step up from what you can do with a raw template.